pkg/resources/schemas/template_test.go

```diff
@@ -0,0 +1,375 @@
+
+				// return updated schemas for the second request
+				factory.EXPECT().Schemas(userInfo).DoAndReturn(func(_ user.Info) (*types.APISchemas, error) {
+					// signal that initial Schemas were called
+					close(eventsReady)
+					return types.EmptyAPISchemas(), nil
+				})
+
+				return testValues{nil, factory, eventsReady}
+			},
+		},
+		{
+			name: "Schema kind attribute is updated",
+			setup: func(ctrl *gomock.Controller) testValues {
+				factory := schemafake.NewMockFactory(ctrl)
+				eventsReady := make(chan struct{})
+				baseSchemas := types.EmptyAPISchemas()
+				updateSchemas := types.EmptyAPISchemas()
+
+				testSchema := types.APISchema{
+					Schema: &v1schema.Schema{
+						ID:                "pod",
+						PluralName:        "pods",
+						CollectionMethods: []string{"GET"},
+						ResourceMethods:   []string{"GET"},
+					},
+				}
+				baseSchemas.AddSchema(testSchema)
+				// initial schemas
+				factory.EXPECT().Schemas(userInfo).Return(baseSchemas, nil)
+
+				// add kind attribute
+				attributes.SetKind(&testSchema, "newKind")
+				updateSchemas.AddSchema(testSchema)
+
+				// return updated schemas for the second request
+				factory.EXPECT().Schemas(userInfo).DoAndReturn(func(_ user.Info) (*types.APISchemas, error) {
+					// signal that initial Schemas were called
+					close(eventsReady)
+					return updateSchemas, nil
+				})
+
+				expectedEvents := []types.APIEvent{
+					{
+						Name:         types.ChangeAPIEvent,
+						ResourceType: "schema",
+						Object: types.APIObject{
+							Type:   resourceType,
+							ID:     testSchema.ID,
+							Object: &testSchema,
+						},
+					},
+				}
+				return testValues{expectedEvents, factory, eventsReady}
+			},
+		},
+		{
+			name: "Schema access attribute is updated",
+			setup: func(ctrl *gomock.Controller) testValues {
+				factory := schemafake.NewMockFactory(ctrl)
+				eventsReady := make(chan struct{})
+				baseSchemas := types.EmptyAPISchemas()
+				updateSchemas := types.EmptyAPISchemas()
+
+				testSchema := types.APISchema{
+					Schema: &v1schema.Schema{
+						ID:                "pod",
+						PluralName:        "pods",
+						CollectionMethods: []string{"GET"},
+						ResourceMethods:   []string{"GET"},
+					},
+				}
+				baseSchemas.AddSchema(testSchema)
+				// initial schemas
+				factory.EXPECT().Schemas(userInfo).Return(baseSchemas, nil)
+
+				// add access attribute
+				attributes.SetAccess(&testSchema, map[string]string{"List": "*"})
+				updateSchemas.AddSchema(testSchema)
+
+				// return updated schemas for the second request
+				factory.EXPECT().Schemas(userInfo).DoAndReturn(func(_ user.Info) (*types.APISchemas, error) {
+					// signal that schemas were requested
+					close(eventsReady)
+					return updateSchemas, nil
+				})
+
+				expectedEvents := []types.APIEvent{}
+				return testValues{expectedEvents, factory, eventsReady}
+			},
+		},
+	}
+
+	for i := range tests {
+		test := tests[i]
```

Comment:
```suggestion
	for _, test := range tests {
```

Reply:
This is similar to my comment above, except the issue with using `_, test` here is that we are using t.Parallel(), which causes a race condition because the same `test` variable is scoped to the loop, not each iteration. The alternative is to add `test := test` at the top of the loop, but I don't like shadowing previously declared variables.